src/main/scala/temple/builder/DatabaseBuilder.scala

```diff
@@ -0,0 +1,57 @@
+package temple.builder
+
+import temple.DSL.semantics.{Annotation, Attribute, AttributeType, ServiceBlock}
+import temple.generate.database.ast.ColumnConstraint.Check
+import temple.generate.database.ast._
+
+/** Construct database queries from a Templefile structure */
+object DatabaseBuilder {
+
+  private def generateMaxMinConstraints[T](name: String, max: Option[T], min: Option[T]): Seq[ColumnConstraint] = {
+    val maxCondition = max.map(v => Check(name, ComparisonOperator.LessEqual, v.toString))
+    val minCondition = min.map(v => Check(name, ComparisonOperator.GreaterEqual, v.toString))
+    Seq(maxCondition, minCondition).flatten
+  }
+
+  private def toColDef(name: String, attribute: Attribute): ColumnDef = {
+    val valueConstraints = attribute.valueAnnotations.map {
+      case Annotation.Unique => ColumnConstraint.Unique
+    }.toSeq
+
+    val (colType, constraints) = attribute.attributeType match {
+      case AttributeType.BoolType     => (ColType.BoolCol, valueConstraints)
+      case AttributeType.DateType     => (ColType.DateCol, valueConstraints)
+      case AttributeType.DateTimeType => (ColType.DateTimeTzCol, valueConstraints)
+      case AttributeType.TimeType     => (ColType.TimeCol, valueConstraints)
+      case AttributeType.ForeignKey   => (ColType.IntCol(4), valueConstraints)
+      case AttributeType.BlobType(_)  => (ColType.BlobCol, valueConstraints)
```

Comment:
Is there a reason the blob size is not enforced?

Reply:
I know the Postgres implementation doesn’t use a length, so it was never added

Reply:
https://github.com/jOOQ/jOOQ/issues/6289#issuecomment-304618975

You can `CHECK` the `octet_size`

Reply:
Also, [jOOQ](https://github.com/jOOQ/jOOQ) looks useful

Reply:
Though they haven’t got around to implementing that yet

Reply:
Nice, shall add that